### PR TITLE
ShouldExistInOrderTest.kt (#30)

### PR DESCRIPTION
**Pull request by AlexCue987**
provide more details when `shouldExistInOrder` fails. For instance, print out "Predicate number 1 did not match", as shown in the following example:
```
        "fail with clear message" {
             shouldThrowAny {
                listOf(1, 2).shouldExistInOrder(
                   { i: Int -> i < 2 },
                   { i: Int -> i < 2 }
                )
             }.message shouldBe "[1, 2] did not match the predicates [(kotlin.Int) -> kotlin.Boolean, (kotlin.Int) -> kotlin.Boolean] in order. Predicate at index 1 did not match."
          }
```

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import io.kotest.matchers.collections.shouldExistInOrder
+import io.kotest.matchers.shouldBe
+
+class ShouldExistInOrderTest: WordSpec() {
+   init {
+       "shouldExistInOrder" should {
+          "pass" {
+              listOf(1, 2).shouldExistInOrder(
+                 { i: Int -> i < 2 },
+             { i: Int -> i < 3 }
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt]: adjust indent

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+              )
+          }
+          "fail with clear message" {
+             shouldThrowAny {
+                listOf(1, 2).shouldExistInOrder(
+                   { i: Int -> i < 2 },
+                   { i: Int -> i < 2 }
+                )
+             }.message shouldBe "[1, 2] did not match the predicates [(kotlin.Int) -> kotlin.Boolean, (kotlin.Int) -> kotlin.Boolean] in order. Predicate number 1 did not match."
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt]: is it the 1st predicate, or the predicate at index 1, that failed?

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt (hunk 1 above)
Comment (AlexCue987) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt]: > adjust indent

done

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt (hunk 2 above)
Comment (AlexCue987) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/ShouldExistInOrderTest.kt]: changed `Predicate number 1` to `Predicate at index 1`

**Review by Kantis — APPROVED** [review]

**Pull request merged**